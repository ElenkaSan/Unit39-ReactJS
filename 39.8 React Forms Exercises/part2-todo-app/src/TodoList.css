.TodoList {
    color: rgb(255, 110, 20);
    font-weight: bold; 
    font-size: 30px; 
}
