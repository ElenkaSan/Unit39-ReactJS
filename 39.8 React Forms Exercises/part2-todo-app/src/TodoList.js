import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuid } from "uuid";
import './TodoList.css';

const TodoList = () => {
    const [lists, setLists] = useState([]);

    const addList = lists => {
        let newList = { ...lists, id: uuid() };
        setLists(lists => [...lists, newList]);
    };

    const removing = (idList) => {
        setLists(lists.filter(({ id }) => id !== idList));
    };

    return (
        <div>
             <h3 className="TodoList">My Todo List</h3>
          <NewTodoForm addingList={addList} />
          {lists.map(({ id, todo, discription}) =>
          <Todo remove={removing} key={id} id={id} todo={todo} discription={discription} />
          )}
        </div>
      )
   };
   
export default TodoList;
   