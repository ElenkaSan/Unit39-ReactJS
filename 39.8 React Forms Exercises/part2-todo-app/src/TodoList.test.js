import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function() {
  render(<TodoList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new todo and delete", function () {
    const { queryByText, getByPlaceholderText } = render(<TodoList />);
    const input = getByPlaceholderText('todo');
    const btn = queryByText("Add Todo!");
    expect(input).toBeEmpty; 
    fireEvent.change(input, { target: { value: "New Todo" } });
    fireEvent.click(btn);
    expect(queryByText(`New Todo`)).toBeInTheDocument;

    const removeBtn = queryByText('X');
    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
    expect(input).toBeEmpty; 
})
