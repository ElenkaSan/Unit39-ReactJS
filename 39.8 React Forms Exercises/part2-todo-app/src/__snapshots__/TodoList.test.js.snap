// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="TodoList"
    >
      My Todo List
    </h3>
    <form>
      <label
        for="Todo"
        style="margin: 10px;"
      >
        Todo:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="todo"
        type="text"
        value=""
      />
      <label
        for="discription"
        style="margin: 10px;"
      >
        Discriptiont:
      </label>
      <input
        id="discription"
        name="discription"
        type="text"
        value=""
      />
      <button
        id="newTodo"
        style="margin: 20px; cursor: pointer; color: rgb(255, 110, 20);"
      >
        Add Todo!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
